c10/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+project(c10 LANGUAGES CXX)
+
+set(LIB_SOURCES
+        dummy.cpp
+)
+
+set(TEST_SOURCES
+        dummy_test.cpp
+)
+
+if("${CMAKE_CXX_COMPILER_ID}" MATCHES "MSVC")
```

Comment:
Why can't we reuse the warning code in the existing main project?

Reply:
You can just use

```
if (MSVC)
...
```
and I would prefer that we don't reset warnings here. Can we try and make this as minimal as possible?

Reply:
Warnings from the main projects should automatically be applied to c10 as well. I'm adding additional warnings because I think this is an opportunity to have new code adhere to stricter quality standards. Ideally, these warnings should be added to the whole repository, but it'd be some effort to fix all of our code, so that's probably going to happen later.